ESP32/DIY-Flow-Bench/hardware.cpp

```diff
@@ -93,27 +106,44 @@ void Hardware::configurePins () {
   
 
   // Outputs
+  #ifdef VAC_BANK_1_PIN
   pinMode(VAC_BANK_1_PIN, OUTPUT);
+  #endif
+
+  #ifdef VAC_BANK_2_PIN
   pinMode(VAC_BANK_2_PIN, OUTPUT);
+  #endif
+
+  #ifdef VAC_BANK_3_PIN
   pinMode(VAC_BANK_3_PIN, OUTPUT);
-  
+  #endif
+
+  #ifdef VAC_SPEED_PIN
   pinMode(VAC_SPEED_PIN, OUTPUT);
+  #endif
+
+  #ifdef VAC_BLEED_VALVE_PIN
   pinMode(VAC_BLEED_VALVE_PIN, OUTPUT);
+  #endif
 
+  #ifdef AVO_STEP_PIN
   pinMode(AVO_STEP_PIN, OUTPUT);
+  #endif
+
+  #ifdef AVO_DIR_PIN
   pinMode(AVO_DIR_PIN, OUTPUT);
+  #endif
 
+  #ifdef FLOW_VALVE_STEP_PIN
   pinMode(FLOW_VALVE_STEP_PIN, OUTPUT);
-  pinMode(FLOW_VALVE_DIR_PIN, OUTPUT);
```

Comment:
Will this need a separate #ifdef FLOW_VALVE_DIR_PIN as it is different to the STEP pin

Reply:
The stepper stuff could all be grouped together, as you would not use one without the other, however this would require that there is a common setting in configuration.h that could then be used to mask out the entire block of stepper related I/O. But at this stage the stepper stuff is all in the future, so there is no clear direction on what any of that looks like until that development starts.

So, I think to maintain continuity, lets add #ifdefs to each I/O point even if it is not currently used in the code.

Reply:
OK 👍